hazelcast/src/test/java/com/hazelcast/config/ConfigTest.java

```diff
@@ -252,18 +255,97 @@ public void testProgrammaticConfigGetUrlAndGetFileReturnNull() {
         assertNull(config.getConfigurationFile());
     }
 
+    @Test
+    public void testMergePersistenceAndHotRestartPersistenceXml() {
+        String dir = randomString();
+        int parallelism = Math.abs(dir.hashCode());
+
+        String xmlTemplate = HAZELCAST_START_TAG
+                + "<persistence enabled=\"%s\">"
+                + "    <base-dir>" + dir + "</base-dir>"
+                + "    <parallelism>" + parallelism + "</parallelism>"
+                + "</persistence>\n"
+                + "<hot-restart-persistence enabled=\"%s\">"
+                + "    <base-dir>" + (dir + "a") + "</base-dir>"
+                + "    <parallelism>" + Math.max(parallelism - 1, parallelism + 1) + "</parallelism>"
+                + "</hot-restart-persistence>\n"
+                + HAZELCAST_END_TAG;
+
+        boolean[][] cases = { {true, true}, {true, false}, {false, true}, {false, false} };
```

Comment:
Minor: we can use parameterized JUnit tests for the same and then each case is reported as a different test case.